python/lsst/sims/catUtils/utils/alertGenerator.py

```diff
@@ -0,0 +1,1475 @@
+        flux = self._dummy_sed.fluxFromMag(mag)
+        dflux = flux - quiescent_flux
+
+        snr_tot, gamma = calcSNR_m5(mag, self.lsstBandpassDict[self.obs_metadata.bandpass],
+                                    self.obs_metadata.m5[self.obs_metadata.bandpass],
+                                    self.photParams, gamma=self._gamma)
+
+        if self._gamma is None:
+            self._gamma = gamma
+
+        return np.array([flux, dflux, snr_tot])
+
+
+class AlertStellarVariabilityCatalog(_baseAlertCatalog,
+                                     VariabilityStars,
+                                     AstrometryStars):
+
+    @compound('quiescent_lsst_u', 'quiescent_lsst_g', 'quiescent_lsst_r',
+              'quiescent_lsst_i', 'quiescent_lsst_z', 'quiescent_lsst_y')
+    def get_quiescent_lsst_magnitudes(self):
+        return np.array([self.column_by_name('umag'), self.column_by_name('gmag'),
+                         self.column_by_name('rmag'), self.column_by_name('imag'),
+                         self.column_by_name('zmag'), self.column_by_name('ymag')])
+
+
+class AlertAgnVariabilityCatalog(_baseAlertCatalog,
+                                 VariabilityGalaxies,
+                                 AstrometryGalaxies):
+
+    @compound('quiescent_lsst_u', 'quiescent_lsst_g', 'quiescent_lsst_r',
+              'quiescent_lsst_i', 'quiescent_lsst_z', 'quiescent_lsst_y')
+    def get_quiescent_lsst_magnitudes(self):
+        return np.array([self.column_by_name('u_ab'), self.column_by_name('g_ab'),
+                         self.column_by_name('r_ab'), self.column_by_name('i_ab'),
+                         self.column_by_name('z_ab'), self.column_by_name('y_ab')])
+
+
+
+class AlertDataGenerator(object):
+    """
+    This class will read in astrophysical sources and variability
+    models from CatSim, observe them with a simulated OpSim
+    cadence, and write a series of sqlite files containing all
+    of the simulated observations that could trigger an alert.
+
+    In order to make this calculation as efficient as possible,
+    the class works by partitioning the sky according to the
+    Hierarchical Triangular Mesh (HTM) of
+
+    Kunszt P., Szalay A., Thakar A. (2006) in "Mining The Sky",
+    Banday A, Zaroubi S, Bartelmann M. eds.
+    ESO Astrophysics Symposia
+    https://www.researchgate.net/publication/226072008_The_Hierarchical_Triangular_Mesh
+
+    Szalay A. et al. (2007)
+    "Indexing the Sphere with the Hierarchical Triangular Mesh"
+    arXiv:cs/0701164
+
+    and simulating all of the observations in a given trixel (the
+    elementary unit of the HTM) at once.  Accordintly, the outputs
+    produced by this class are files named like
+
+    prefix_NNNN_sqlite.db
+
+    where prefix is specified by theuser and NNNN is the htmid, the
+    unique identifying integer, corresponding to each simulated trixel.
+
+    The proper way to run this class is to instantiate it, run
```

Comment:
As soon as the code tries to load the Kepler light curve file, it will raise the exception here

https://github.com/lsst/sims_catUtils/blob/master/python/lsst/sims/catUtils/mixins/VariabilityMixin.py#L909

if the file doesn't exist.  Since loading the Kepler file is the first step in `AlertDataGenerator.__init__()`, it should trigger early enough that people won't waste time on a process that is going to crash anyway.

Do you disagree that this is safe enough?

Reply:
No, that sounds good.